packages/api-clients/api-codegen-preset/src/helpers/get-schema-data.ts

```diff
@@ -1,14 +1,42 @@
-import type {ApiType} from '../types';
+import {ApiType} from '../types';
 
 import {apiConfigs} from './api-configs';
 
 export function getSchemaData(
   outputDir: string,
   apiType: ApiType,
   apiVersion?: string,
+  apiKey?: string,
 ) {
   const config = apiConfigs[apiType];
 
+  if (apiType === ApiType.Customer) {
+    if (!apiKey) {
+      throw new Error('The customer API requires an API key');
+    }
+
+    const schemaUrl = (
+      apiVersion
+        ? config.schema.replace('%%API_VERSION%%', apiVersion)
+        : config.schema.replace('&api_version=%%API_VERSION%%', '')
+    ).replace('%%API_KEY%%', apiKey);
```

Comment:
Took me a second to get this - could we replace each of the values in a separate step to make it more readable?